exaudfclient/docs/script_options_design.md

```diff
@@ -0,0 +1,466 @@
+For white space character within an option value, those must be added as is to the result option value.
+For white space character in anything else, what is not an option, those white space characters must be ignored.
+
+
+
+Covers:
+- `req~white-spaces-script-options-parsing-v2~1`
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Lexer and Parser Rules Whitespace Escape Sequences
+`dsn~lexer-parser-rules-whitespace-escape-sequences~1`
+
+Define the Lexer rules to tokenize whitespace escape sequences:
+- '\ ' => space character
+- '\t' => <tab> character
+- '\f' => <form feed> character
+- '\v' => <vertical tab> character
+
+Implement rules which replace those white space escape tokens only at the beginning of an option value. Add parser rules to ignore those tokens in anything else, which is not a script option. Those token are not expected to be part of an option key.
+
+
+Covers:
+- `req~leading-white-spaces-script-options-parsing~1`
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Lexer and Parser Rules Escape Sequences
+`dsn~lexer-parser-rules-escape-sequences~1`
+
+Define the Lexer rules to tokenize '\n', '\r', '\; sequences, and parser rules to replace those sequences with <line feed>, <carriage return> or ';' characters. 
+Implement rules which replace those token at any location in an option value. Add parser rules to ignore those tokens in anything else, which is not a script option. Those token are not expected to be part of an option key.
+
+
+Covers:
+- `req~escape-sequence-script-options-parsing~1`
+Depends:
+- `dsn~parser-implementation~1`
+
+Tags: V2
+
+### Script Option Removal Mechanism
+`dsn~script-option-removal~1`
+
+Implement a method in class `ScriptOptionLinesParserCTPG` which removes *all* identified Script Options from the original script code at once. This method be executed after the import scripts are replaced. The algorithm must replace the script options in reverse order in order to maintain consistency of internal list of positions.
+
+
+Covers:
+- `req~script-option-removal~1`
+- `req~java-scriptclass-option-handling-v2~1`
+
+Tags: V2
+
+### Unknown Options Check
+`dsn~script-option-unknown-options-behvaior-v2~1`
+
+Implement a check in class `ScriptOptionLinesParserCTPG` which verifies that only known script options are found. Otherwise, raise an exception with information about the position and name of the unknown option.
+
+Covers:
+- `req~script-option-unknown-options-behavior-v2~1`
+
+Tags: V2
+
+### Java %scriptclass Option Handling in Design
+`dsn~java-scriptclass-option-handling~1`
+
+Implement a function in the `Converter` class which adds the script class option to the JVM Options list.
+
+
+Covers:
+- `req~java-scriptclass-option-handling-v2~1`
+
+Tags: V2
+
+### Java %jar Option Collection
+`dsn~java-jar-option-collection~1`
+
+Implement an algorithm in class `ConverterV2` to split the given Jar option value by colon (':') and then collect the found Jar options in a list. 
```

Comment:
Add our links about what are legal classpaths

Reply:
Done

Reply:
someone might not read the requirement and only the design

Reply:
Added the link